### ISSUE TITLE
Representing recommended fields in the JSON schema

### ISSUE THREAD

**Issue by ekoner** [ThreeSixtyGiving/standard#180]
As part of the State of the Data reporting, Katherine needs to know which sections and fields are being used in the 360Giving standard. Not all sections and fields are equally important, so at the moment, we have two possibilities in the schema: `Optional` and `Required`.

From our work on the data quality tool and ongoing support, we know that some sections or fields are `Recommended`, for example `Last Modified` and `Data Source`. These aren't represented in the schema. 

**How can we represent `Recommended` sections and fields in the JSON Schema?**

This would ensure that:

- Once agreed with the stewardship committee, they are SSOT. 
- They can be re-used on the website and in the data quality tool. 
- We could look at this as part of the project to improve the documentation around 360Giving.

**Comment by timgdavies** [comment]
This is really interesting issue - and something I think that might cut across all the standards we work on with JSON schema. 

To bring in learning from OCDS (at fear of complicating things a bit) there might be a concept of 'recommended **for**'. I.e. 'This field is recommended **if** you are targeting basic publication, or recommended **if** you want to be included in GrantNav. 

I've had a quick scan to see if [JSON Schema (or forthcoming version)](https://github.com/json-schema-org/json-schema-spec/issues) provides any such properties but I don't believe it does. 

If we do mint something new here, then we might want to start maintaining our own set of patches to the JSON Schema meta-schema, so we have a way of clearly documenting the extensions we're making to it.

**Comment by stevieflow** [comment]
@robredpath I think there's been some work on this recently?

**Comment by mrshll1001** [comment]
I'm tentatively closing this issue.

My reasoning is that *all* of the fields in the Standard are recommended, unless they are explicitly marked as deprecated. The idea being we want to  live in a world where publishers can and do produce as much good quality data about their grantmaking as possible. The act of adding a field into the Standard is a recommendation that this field should be in datasetes if possible.

This of course differs from the minimum viable set of fields which make data about a grant useful at all, which we encapsulate as the required fields of the Standard.

I agree with the approach that beyond the minimum, we should think about specific fields as being *recommended or necessary for particular types of use*. I think the challenge is therefore to come up with a way of documenting use cases against fields in the Standard, so that data quality tooling can determine whether you can use the data for X or Y particular use case.

**Comment by mariongalley** [comment]
@mrshll1001 Good point about aligning to use-cases. It wouldn't be that hard for the DQT to say "Your data can't be used to make maps and contribute to identifying duplication and cold spots" rather than "Your data doesn't have geocodes"
